src/main/scala/com/krykowski/service/TaskServiceImpl.scala

```diff
@@ -0,0 +1,100 @@
+package com.krykowski.service
+
+import cats.effect.{Blocker, IO}
+import com.krykowski.model
+import com.krykowski.model.{CsvFileLocation, Done, Running, Scheduled, Task, TaskNotFoundError}
+import com.krykowski.repository.TaskRepository
+import com.softwaremill.sttp.{Response, TryHttpURLConnectionBackend, UriContext, sttp}
+import io.circe.generic.auto.exportEncoder
+import io.circe.syntax.EncoderOps
+import org.slf4j.LoggerFactory
+
+import java.io.File
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.language.postfixOps
+import sys.process._
+import java.net.URL
+import java.nio.file.{Path, Paths}
+
+class TaskServiceImpl(taskRepository: TaskRepository) extends TaskService {
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  implicit private val cs = IO.contextShift(global)
+  implicit private val cf = IO.ioConcurrentEffect
+
+  override def createTask(csv: CsvFileLocation): IO[Long] = {
+    getCsvFile(csv.uri) match {
```

Comment:
File is downloaded without checking if other tasks are running. Calling thread is blocked. The is no way how to cancel a running task at this stage.